src/query-builder/select-query-builder.ts

```diff
@@ -84,6 +85,28 @@ import { parseFetch } from '../parser/fetch-parser.js'
 import { TopModifier } from '../operation-node/top-node.js'
 import { parseTop } from '../parser/top-parser.js'
 
+type RemoveIndex<T> = keyof {
```

Comment:
this is required so we don't Omit index types, e.g. `{[x: string]: unknown}`.
I did this to preserve the integrity of the unit test here: https://github.com/kysely-org/kysely/blob/ec68d6bd0b2da7899a169b3f89e0c29bbaf1bf7c/test/typings/test-d/select.test-d.ts#L39-L43.

If performance is an issue, we could modify that test & remove this